error.go

```diff
@@ -99,9 +134,21 @@ func (g *GraphQLError) Response() *http.Response {
 	return g.response
 }
 
+func (g *GraphQLError) Code() string {
+	errors := g.errors
+
+	if len(errors) > 0 {
+		return errors[len(errors)-1].ErrCode()
```

Comment:
Are graphql codes different than http codes status code? If so, the return of this function may be of """""""varying types""""""", which would reduce the interchangeability (is this even a word?) between our error types by their handlers.

My suggestion: Create a "code type" specific to our lib (it could be the graphql codes, http codes or even a new one) so that our callers can use it without needing to relly on some sort of reflection to check the error type/struct.

PS: This doesn't need to be done in this PR, it could very well be done in another one since it would be a very clearly breaking change.